### PR TITLE
EQL: Skip execution for filters with empty results

### LINKED ISSUE #53237: EQL: Folding condition optimized to false does not return an empty result
The optimizer is aware that the `where` clause optimized to `false` returns no results however the implementation is not fully implemented.
The expression is resolved into a `LocalRelation` however there's no execution plan for it leading to an exception.

### PR DESCRIPTION
**Pull request by costin**
Optimize away events queries and joins/sequence that cannot match any
results without having to query Elasticsearch

Fix #53237

### CODE REVIEW

File: x-pack/plugin/eql/qa/common/src/main/resources/test_queries_supported.toml [hunk 1]
```diff
@@ -146,4 +146,12 @@ sequence
...
 '''
-expected_event_ids  = [1, 2, 2, 3]
\ No newline at end of file
+expected_event_ids  = [1, 2, 2, 3]
+
+[[queries]]
+query = '''
+sequence
+  [process where false] by unique_pid
+  [process where true] by unique_ppid
```
Comment (matriv) [inline x-pack/plugin/eql/qa/common/src/main/resources/test_queries_supported.toml:155]: @costin Question: if we have let's say 4 queries (stages) and one of them (no matter which) is no_match (filter => false) then we cannot match any sequence, correct?

Comment (costin) [reply x-pack/plugin/eql/qa/common/src/main/resources/test_queries_supported.toml:155]: Yes, hence the test and rule in this PR.

**Comment by costin** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-unix-archives

**Review by matriv — APPROVED** [review]
LGTM

**Review by astefan — APPROVED** [review]
LGTM

**Pull request merged**